packages/react/src/runtime/ax.ts

```diff
@@ -41,7 +48,15 @@ export default function ax(classNames: (string | undefined | null | false)[]): s
       continue;
     }
 
-    const groups = cls.split(' ');
+    let groups: string[] = [];
+    if (typeof cls === 'string') {
+      groups = cls.split(' ');
+    } else if (Array.isArray(cls)) {
+      groups = cls; // NOTE: Not really a valid scenario, so we don't deeply recurse or handle this properly
+    } else {
+      // NOTE: Could throw an error here if this isn't our `AtomicGroups` to help understand that this is collecting invalid styles.
```

Comment:
I found doing a `cls instanceof AtomicGroups` wasn't consistently working, might as well try to blanket support, though…it does mean previous things that would throw `'cls.split is not a function'` now more quietly fail.

Reply:
Yeah that's fine. We can lean on the typesystem for everything else.

Reply:
do you guys use `ac` and `AtomicGroups` for anything? i thought that was part of the [class name compression work](https://github.com/atlassian-labs/compiled/pull/1437) that we've now shelved (probably safe to just yeet)

Reply:
So the root example comes from using `cx()` which uses `ac()`.  Frankly, I do not know _anything_ here, I'm just hacking it together.

Reply:
I think we used `ac` based on @liamqma's guidance. If you think it's not useful we can switch it out for whatever.

Reply:
[ahh i see](https://github.com/atlassian-labs/compiled/pull/1533/files#diff-2821c097864f8f4f268973d4fafa1cab339af4bd3805b489c4777a0cb8ff0294R144)

i forgot we were using it there haha, thank you for clarifying

Reply:
Yeah. So should we just use ax and side step the whole issue?

Reply:
Yeah, we've parked classname compression. In theory, no Compiled users should use `ac`. 
I suggest switching it out for the old good `ax`. 🙂

Reply:
oops i was going to write a big comment but it got deleted 😢 

but yeah depends on whether you want to support nesting like `cx([cx('_aaaa_a'), '_aaaa_b'])`... if not, i guess there's no value that an `ac`-based approach would provide

Reply:
(and maybe delete the `ac`/class name compression stuff altogether too? but that's probably more suited for a follow-up PR)

Reply:
Makes sense to me, is it literally as simple as swapping out `ac` for `ax`?  It kinda looks like it locally as I _think_ I've got all my tests passing 🙈

If that's a drop-in, I can pop that out tomorrow, but I won't make the whole removal (bit over my head).

Reply:
I _think_ this works: https://github.com/atlassian-labs/compiled/pull/1697